consensus/wal.go

```diff
@@ -229,12 +229,17 @@ func NewWALEncoder(wr io.Writer) *WALEncoder {
 	return &WALEncoder{wr}
 }
 
-// Encode writes the custom encoding of v to the stream.
+// Encode writes the custom encoding of v to the stream. It returns an error if
+// the amino-encoded size of v is greater than 1MB. Any error encountered
+// during the write is also returned.
 func (enc *WALEncoder) Encode(v *TimedWALMessage) error {
 	data := cdc.MustMarshalBinaryBare(v)
 
 	crc := crc32.Checksum(data, crc32c)
 	length := uint32(len(data))
+	if length > maxMsgSizeBytes {
+		return fmt.Errorf("Msg is too big: %d bytes, max: %d bytes", length, maxMsgSizeBytes)
```

Comment:
How would this ever happen? Afaik it never should for an honest node. If this happened for someone seems we might have a different issue?

Reply:
> If this happened for someone seems we might have a different issue?

Maybe, but I don't think we can get more feedback right now. I don't know the nickname, nor he/she provided more info back then.

Nevertheless, this check makes sense for me.